### DISCUSSION TITLE
ArgoCD behavior on chart repo outage

### DISCUSSION THREAD

**Discussion by ITler** [argoproj/argo-cd#9297] [General]
Dear ArgoCD team,

I am wondering how ArgoCD would behave, when it cannot sync an application because the application `source` is not reachable, because i.e. the source system is down or does not serve as expected.

Would ArgoCD be able to distinguish between not being able to read and process existing helm chart sources and resources not being there by definition means not defined by intention? 

I wouldn't want ArgoCD to delete already deployed resources just because it cannot read the deploy code (helm charts) from `source`. Is this mitigated by ArgoCD?

**Comment by chetan-rns** [comment]
Hi @ITler, Argo CD might show a `404: Not Found Error` if the source is not found. However, I don't think it'll delete already deployed resources.

**Reply by ITler** [reply]
Hello @chetan-rns,

thanks for following up on my question, much appreciated. Yeah, showing the 404 would absolutely make sense. Still, and please don't get me wrong with that, may I kindly ask, do you **think** or do you **know** that Argo would never delete resources if the source wouldn't serve or maybe even only partially serve.
It is quite hard for me to discover the lines in code to answer the question by myself, because I am not familiar with argo's code structure. It is a lot of code :slightly_smiling_face:

**Reply by rishabh625** [reply]
@ITler it doesn't delete resources if source is not available and reconcilation is run
